sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -1399,6 +1401,7 @@ CACHE: 'CACHE';
 CASCADE: 'CASCADE';
 CASE: 'CASE';
 CAST: 'CAST';
+CATALOG: 'CATALOG';
```

Comment:
We also need to put it in `ansiNonReserved` and `nonReserved`

Reply:
If we put it in `nonReserved`, `USE CATALOG` is parsed as `USE namespace=multipartIdentifier`. Any trick to go around this?

Reply:
this is a good question. `USE abc` has been supported for a long time and we can't drop it.

I think the extended `USE ... IN ...` syntax is already very powerful, the only missing case is to switch to a catalog and its default namespace.

Since we need a new syntax for it anyway, how about `SET CURRENT CATALOG ...`?

Reply:
`SET CURRENT CATALOG ...` will be parsed as `setConfiguration` which is `SET .*? ` :)

Reply:
Actually, putting `USE CATALOG` before `USE` seems to work fine now even if I set it to `ansiNonReserved` and `nonReserved`. I will do a bit more testing and update this thread if there is any issue.